### ISSUE TITLE
Cannot run tests without FreeImage.

### ISSUE THREAD

**Issue by ghisvail** [arrayfire/arrayfire#1165]
I tried to rebuild the packaging without FreeImage since the imageio testsuite is failing on some architectures which you guys don't officially support. The testsuite builds successfully but running it produces the following outcome:
```
The following tests FAILED:
	  7 - Test_fast_cpu (Failed)
	 23 - Test_susan_cpu (Failed)
	 44 - Test_orb_cpu (Failed)
	 50 - Test_homography_cpu (Failed)
	 51 - Test_harris_cpu (Failed)
	 76 - Test_morph_cpu (Failed)
	 80 - Test_bilateral_cpu (Failed)
	 84 - Test_meanshift_cpu (Failed)
```

**Comment by shehzan10** [comment]
I tried doing this when I was separating out the test. Unfortunately there is no neat way to do it.

Here are the options:

1. We remove each test manually (like we do for sift/manual_memory) in CMake. The problem with this is that each time a new test is added that uses image io, it will need to be added here. The solution to this would be to make it like testname_imageio.cpp and then we can look for imageio to disable. As I said, not a nice way.
2. The second option is to have a `#ifndef WITH_FREEIMAGE return; #endif` block in each test that uses image io. This is more similar to the `if (noDoubleTests<T>()) return;` mechanism.

**Comment by umar456** [comment]
3. We could get rid of the dependency on free image for those tests. Create a PFM of the images used by those tests and create a simple loader (I have one here somewhere). That way we don't have to unnecessarily disable any tests.

**Comment by pavanky** [comment]
Using PFM images is going to blow up the size of the arrayfire-data repo. I highly recommend against it.

**Comment by umar456** [comment]
That is better then not running tests

**Comment by 9prady9** [comment]
Is this for package on some OS's package manager ?

**Comment by pavanky** [comment]
@umar456 @9prady9 That's my point. We test this with freeimage in jenkins so I don't see the point of enabling tests without freeimage.

**Comment by 9prady9** [comment]
I don't see the need either if this is for just packaging purposes.

**Comment by shehzan10** [comment]
Merged into hotfixes-3.2.2
